controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -277,48 +273,59 @@ private ApplicationController applications() {
                                     jobs.add(deploymentJob(application, versions, change, job, reason, completedAt.get()));
-                                if ( ! alreadyTriggered(application, versions))
+                                }
+                                if (!alreadyTriggered(application, versions)) {
                                     testJobs = emptyList();
-                            }
-                            else if (testJobs == null) {
-                                testJobs = testJobs(application, versions, String.format("Testing deployment for %s (%s)", job.jobName(), versions.toString()),
-                                                    completedAt.orElse(clock.instant()));
+                                }
+                            } else if (testJobs == null) {
+                                testJobs = testJobs(application, versions,
+                                                    String.format("Testing deployment for %s (%s)",
+                                                                  job.jobName(), versions.toString()),
+                                                    completedAt.orElseGet(clock::instant));
                             }
                         }
                         completedAt = Optional.empty();
-                    }
-                    else { // All jobs are complete; find the time of completion of this step.
+                    } else { // All jobs are complete; find the time of completion of this step.
                         if (stepJobs.isEmpty()) { // No jobs means this is delay step.
                             Duration delay = ((DeploymentSpec.Delay) step).duration();
-                            completedAt = completedAt.map(at -> at.plus(delay)).filter(at -> ! at.isAfter(clock.instant()));
+                            completedAt = completedAt.map(at -> at.plus(delay)).filter(at -> !at.isAfter(clock.instant()));
                             reason += " after a delay of " + delay;
-                        }
-                        else {
+                        } else {
                             completedAt = stepJobs.stream().map(job -> application.deploymentJobs().statusOf(job).get().lastCompleted().get().at()).max(naturalOrder());
                             reason = "Available change in " + stepJobs.stream().map(JobType::jobName).collect(joining(", "));
                         }
                     }
                 }
-            if (testJobs == null)
+            }
+            if (testJobs == null) {
                 testJobs = testJobs(application, Versions.from(application, controller.systemVersion()),
                                     "Testing last changes outside prod", clock.instant());
+            }
             jobs.addAll(testJobs);
         });
-        return jobs;
+        return Collections.unmodifiableList(jobs);
+    }
+
+    /** Returns whether given job can be triggered at this point in time */
```

Comment:
This is valid for production jobs only, so I think the doc should reflect that, and possibly also make an assertion at the start of the method.

Reply:
That's also the reason I kept it inlined, but I won't protest against you separating it out again :)